pyomo/contrib/pynumero/sparse/block_vector.py

```diff
@@ -295,256 +310,218 @@ def dot(self, other, out=None):
-        raise NotImplementedError('resize not implemented for BlockVector')
+        assert_block_structure(self)
+        assert out is None, 'Out keyword not supported'
+        return self.max()-self.min()
 
     def round(self, decimals=0, out=None):
         """
-        Return a with each element rounded to the given number of decimals
+        Return BlockVector with each element rounded to the given number of decimals
         """
-        if out is not None:
-            raise NotImplementedError('round not implemented with out input')
+        assert_block_structure(self)
+        assert out is None, 'Out keyword not supported'
         result = BlockVector(self.nblocks)
-        for idx, blk in enumerate(self):
-            if self._block_mask[idx]:
-                result[idx] = blk.round(decimals=0, out=None)
-            else:
-                result[idx] = None
+        for idx in range(self.nblocks):
+            result.set_block(idx, self.get_block(idx).round(decimals=decimals))
         return result
 
-    def searchsorted(self, v, side='left', sorter=None):
-        raise NotImplementedError('searchsorted not implemented for BlockVector')
-
-    def setfield(self, val, dtype, offset=0):
-        raise NotImplementedError('setfield not implemented for BlockVector')
-
-    def setflags(self, write=None, align=None, uic=None):
-        raise NotImplementedError('setflags not implemented for BlockVector')
-
-    def sort(self, axis=-1, kind='quicksort', order=None):
-        raise NotImplementedError('sort not implemented for BlockVector')
-
-    def squeeze(self, axis=None):
-        raise NotImplementedError('squeeze not implemented for BlockVector')
-
     def std(self, axis=None, dtype=None, out=None, ddof=0, keepdims=False):
         """
-        Returns the standard deviation of the array elements along given axis.
+        Returns the standard deviation of the BlockVector elements.
         """
         return self.flatten().std(axis=axis, dtype=dtype, out=out, ddof=ddof, keepdims=keepdims)
 
-    def swapaxes(self, axis1, axis2):
-        raise NotImplementedError('swapaxes not implemented for BlockVector')
-
-    def take(self, indices, axis=None, out=None, mode='raise'):
+    def var(self, axis=None, dtype=None, out=None, ddof=0, keepdims=False):
         """
-        Return an array formed from the elements of a at the given indices.
+        Returns the variance of the BlockVector elements.
         """
-        return self.flatten().take(indices, axis=axis, out=out, mode=mode)
-
-    def tobytes(self, order='C'):
-        raise NotImplementedError('tobytes not implemented for BlockVector')
+        return self.flatten().var(axis=axis, dtype=dtype, out=out, ddof=ddof, keepdims=keepdims)
 
     def tofile(self, fid, sep="", format="%s"):
         """
-        Write array to a file as text or binary (default).
+        Writes flat version of BlockVector to a file as text or binary (default).
```

Comment:
I actually already had to write a method to serialize the structure. This would be pretty easy to do (although it would not really be readable). I will add an issue.